apache/rocketmq/v2/service.proto

```diff
@@ -244,6 +244,63 @@ message ChangeInvisibleDurationResponse {
   string receipt_handle = 2;
 }
 
+message PullMessageRequest {
+  Resource group = 1;
+  MessageQueue message_queue = 2;
+  int64 offset = 3;
+  int32 batch_size = 4;
+  FilterExpression filter_expression = 5;
+  google.protobuf.Duration long_polling_timeout = 6;
+}
+
+message PullMessageResponse {
+  oneof content {
+    Status status = 1;
+    Message message = 2;
+    int64 next_offset = 3;
+  }
+}
+
+message UpdateOffsetRequest {
+  Resource group = 1;
+  MessageQueue message_queue = 2;
+  int64 offset = 3;
+}
+
+message UpdateOffsetResponse {
+  Status status = 1;
+}
+
+message GetOffsetRequest {
+  Resource group = 1;
+  MessageQueue message_queue = 2;
+}
+
+message GetOffsetResponse {
+  Status status = 1;
+  int64 offset = 2;
+}
+
+message QueryOffsetByTimestampRequest {
+  MessageQueue message_queue = 1;
+  google.protobuf.Timestamp timestamp = 2;
+}
+
+message QueryOffsetByTimestampResponse {
+  Status status = 1;
+  int64 offset = 2;
+}
+
+message QueryOffsetByGroupRequest {
```

Comment:
What's the difference between QueryOffsetByGroupRequest and GetOffsetRequest

Reply:
@xdkxlk QueryOffsetByGroupRequest/QueryOffsetByGroupResponse has been removed.